pkg/api/aim/encoding/decoder.go

```diff
@@ -0,0 +1,110 @@
+package encoding
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/rotisserie/eris"
+)
+
+func Decode(data io.Reader) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+	d := decoder{bufio.NewReader(data)}
+	for {
+		key, err := d.readField()
+		if err != nil {
+			if err == io.EOF {
+				return result, nil
+			}
+			return result, eris.Wrap(err, "error reading data line")
+		}
+		var index bool
+		var path []string
+		for _, p := range bytes.Split(key, []byte{0xFE}) {
+			switch {
+			case index:
+				i := int64(binary.BigEndian.Uint64(p))
+				path = append(path, strconv.FormatInt(i, 10))
+				index = false
+			case len(p) == 0:
+				index = true
+			default:
+				path = append(path, string(p))
+			}
+		}
+
+		valuebuf, err := d.readField()
+		if err != nil {
+			return nil, eris.Wrap(err, "error reading data line")
+		}
+
+		var value any
+		switch valuebuf[0] {
+		case 0:
+			value = nil
+		case 1:
+			value = valuebuf[1] != 0
+		case 2:
```

Comment:
Could we use constants instead of magic numbers to make the intention of this switch clearer? ie: 

const INT = 2
const FLOAT = 3 // etc 

case INT16: // then do the length comparisons.

Reply:
I didn't touch the logic itself, because it isn't really in scope of this task. maybe reasonable to do that in separate task. what do you think?

Reply:
ok. done.